CMakeLists.txt

```diff
@@ -126,6 +127,20 @@ else()
     -DGOOGLE_PROTOBUF_NO_RTTI)
 endif(WIN32)
 
+# configure a header file to pass some of the CMake settings
+# to the source code
+
+git_describe(MCSEMA_VERSION_STRING)
+git_get_branch(MCSEMA_BRANCH_NAME)
+
+configure_file (
+  "mcsema/Version.h.in" #input is based on CMAKE_SOURCE_DIR
```

Comment:
Can this be done kind of like how `mcsema/CFG/CFG.h` works, where there's a protobuf header file placed in the `generated` dir, and then it is included by an in-tree header file?